_posts/2020-08-11-pull-requests-require-context.md

```diff
@@ -0,0 +1,160 @@
+bug.
+
+_Bonus tip_: it can be easier to review PRs with many small commits via the
+[_Commits_](https://github.com/artsy/convection/pull/645/commits) tab instead of the
+[_Files changed_](https://github.com/artsy/convection/pull/645/files) tab.
+
+### Demonstrate the results
+
+Pictures are a worth a thousand words. Animated gifs are worth a thousand pictures (uhhhh, in file size too 😬). An
+animated gif showing the outcome of your PR gives reviewers a demo, and confidence that you've verified your
+changes.
+
+[Here's a PR](https://github.com/artsy/force/pull/5817) from [Ashley](https://github.com/ashleyjelks) that includes
+animated gifs of the changes she's made. The effects of the changes might not be obvious by looking only at the
+code, but seeing them in action makes it clear.
+
+### Document the unseen
+
+Sometimes a PR's changes have effects outside of the UI. There are still ways to give reviewers proof that the
+changes have the desired effects.
+
+[Here's a PR](https://github.com/artsy/eigen/pull/3206) from [Yuki](https://github.com/yuki24) that not only
+demonstrates what's happening in the UI, but also assures me that the back-end data is getting updated properly
+through a Rails console.
+
+[Christina](https://github.com/xtina-starr) authored [this PR](https://github.com/artsy/reaction/pull/3441) which
+shows the UI changes in addition to some output from her browser console, demonstrating that analytics tracking
+calls are firing correctly.
+
+### Share your progress
+
+One mistake many engineers make with non-trivial pull requests is to wait to open them until they're "done". If
+there are changes you'd like to get people's eyes on quickly, open a WIP PR before the work is done: mark it as a
+draft in GitHub, or put `WIP` in the title. Extra work up front avoids rework by starting early discussions about
+your approach.
+
+Let reviewers know in the body that your work isn't complete. As you continue your work, use a `TODO:` list in the
+body to illustrate your progress, as in [this PR](https://github.com/artsy/palette/pull/464) from
+[Sepand](https://github.com/sepans).
+
+Is this PR part of a larger scope of work? Is there followup work that will need to be done after it's merged? Are
+there PRs in other systems that need to merge in a specific sequence? Any migration details or timing that should
+be known before merging? Call these details out to avoid another round-trip conversation.
+
+## Spread knowledge
+
+Pull requests should not be one-sided - they aren't just about collecting feedback from the reviewer. They're also
+an opportunity to spread knowledge from the author.
+
+### Share your learnings
+
+Maybe you learned some things about the system you're working with, or you learned a new feature of the language.
+Share this new information with your team. [Roop](https://github.com/anandaroop) shares some findings about
```

Comment:
@anandaroop any objections to me referencing you & your PR here?

Reply:
All good, thanks for asking 💜